docs: document the return type restrictions for operator overloading (#20529)
vlang/v: doc/docs.md

- When overriding `<` and `==`, the return type must be strictly `bool`.
- Both arguments must have the same type (just like with all operators in V).

> - Overloaded operators has to return the same type as the argument (with the expection `<` and `==` which returns boolean expressions).

Reviewer: As this line goes beyond the 100 words. I think

```suggestion
- Overloaded operators have to return the same type as the argument (with the exception `<` and `==).
```

works better

Author: Wow, trying to write to fast hehe :/
